### PR TITLE
Provide types for typescript

### PR DESCRIPTION
**Pull request by glorat**
* Create index.d.ts

* Create vue.d.ts

* Update package.json

### CODE REVIEW

**Comment by glorat** [comment]
Relates to issue https://github.com/pulsardev/vue-tour/issues/73

I've tested this locally against my consuming project but I only use very little of the full API so I've correspondingly kept the exposed interface small and to the methods I'm confident have been declared correctly.

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+export function install (vue: typeof Vue): void
+
+export interface Tour {
+  start(startStep?: string): void
+  previousStep(): void
+  nextStep(): void
+  stop(): void
+  skip(): void
+  finish(): void
+  currentStep: number
```
Comment (Snapu) [inline types/index.d.ts]: Would be nice to also include the computed property `isRunning`. E.g. I use it to ensure my tours do not overlap.

Comment (glorat) [reply types/index.d.ts]: I've added that to the PR now. Please test by putting 
`"vue-tour": "git://github.com/glorat/vue-tour.git#typings",`

in the dependencies section of your package.json and let me know if that works for you

**Comment by HZooly** [comment]
Thanks for this @glorat, we will check the PR with the team as soon as possible!

**Comment by glorat** [comment]
@Snapu and anyone else looking to test (and give feedback!) before final review and merge, you can add this to your dependencies in package.json to test the typings
`"vue-tour": "git://github.com/glorat/vue-tour.git#typings"`

Otherwise, "it works for me" :)

**Comment by Snapu** [comment]
> @Snapu and anyone else looking to test (and give feedback!) before final review and merge, you can add this to your dependencies in package.json to test the typings
> `"vue-tour": "git://github.com/glorat/vue-tour.git#typings"`
> 
> Otherwise, "it works for me" :)

Just tested, it works for me too 👍

File: types/index.d.ts [hunk 2]
```diff
@@ -4,11 +4,18 @@ import Vue from 'vue';
...
   start(startStep?: string): void
   previousStep(): void
   nextStep(): void
   stop(): void
   skip(): void
   finish(): void
   currentStep: number
+
+  // Computed
+  isRunning(): boolean
```
Comment (Snapu) [inline types/index.d.ts]: This is not working. You have to remove `()` for the computed props.
E.g.:
```
isRunning: boolean
```

Comment (glorat) [reply types/index.d.ts]: Thanks for the feedback. I've pushed the update. Let me know if that works better.

Comment (Snapu) [reply types/index.d.ts]: Now it works. I have tested 👍

**Comment by Outpox** [comment]
Thanks for the PR, your investment is greatly appreciated!  
I only have one struggle with it, you haven't followed the commit naming guidelines: https://github.com/pulsardev/vue-tour/blob/master/CONTRIBUTING.md#git-commit-guidelines  
This allows us to easily generate changelogs and have a streamlined history.  
@mmorainville won't be pleased 🤣  

My suggestion here would be: follow the guidelines from now on and we'll rename theses commits (without changing the commits authorship obviously).

**Comment by glorat** [comment]
> Thanks for the PR, your investment is greatly appreciated!
> I only have one struggle with it, you haven't followed the commit naming guidelines: https://github.com/pulsardev/vue-tour/blob/master/CONTRIBUTING.md#git-commit-guidelines
> This allows us to easily generate changelogs and have a streamlined history.
> @mmorainville won't be pleased 🤣
> 
> My suggestion here would be: follow the guidelines from now on and we'll rename theses commits (without changing the commits authorship obviously).

I learned something new today! I should consider adopting the same guidelines in some of my projects... release notes are a pain

I've squashed and renamed my PR

**Comment by Outpox** [comment]
Thanks for your reactivity and sorry to nitpick (😅) but the [subject chapter](https://github.com/pulsardev/vue-tour/blob/master/CONTRIBUTING.md#subject) states: `don't capitalize first letter`

**Comment by glorat** [comment]
> Thanks for your reactivity and sorry to nitpick (😅) but the [subject chapter](https://github.com/pulsardev/vue-tour/blob/master/CONTRIBUTING.md#subject) states: `don't capitalize first letter`

Fixed (and fixed in my other repo where I start applying this standard). Thanks for the education

**Comment by speasley** [comment]
Hello, I'm wondering what the status of this PR is.

**Comment by speasley** [comment]
Is this project no longer maintained?

**Comment by HZooly** [comment]
Hello @speasley, please remember that we're doing this project during our free time.
Thanks to @glorat, types seem to be in a good progress, but we have to review it, test it etc.

We have a meeting tomorrow, we will talk about it 🚀

**Comment by HZooly** [comment]
@glorat @Snapu @speasley [Vue Tour 1.3.1](https://www.npmjs.com/package/vue-tour/v/1.3.1) is published with Types 🥳

**Pull request merged**